File: Sagittarius-Modules/Java/MOTDModule.java
import sagittarius.GetAction;
import sagittarius.Handler;
import sagittarius.Module;
import sagittarius.SagResponse;
import sagittarius.Sagittarius;

public class MOTDModule extends Module {

    private String motd;
    
    private Handler handler;

    public MOTDModule(Sagittarius s) {
        super(s);
        this.ID = "motd";
    }
    
    public void RegisterOnMOTDReceivedHandler(Handler h) {
        this.handler = h;
    }

    public String GetMOTD() {
        return motd;
    }

    public void QueryMOTD() {
        GetAction ga = (GetAction)parent.CreateAction("get");
        ga.AddFilter(GetAction.DBTYPE, "motd");
        ga.AddFilter(GetAction.DBNAME, "motd");
        ga.AddProjection("message", true);
        ga.Unique();
        SubmitAction("motdquery", ga);
    }

    @Override
    public void OnTextReceived(String ActionID, SagResponse resp) {
        motd = resp.getValue("message");
    }

    @Override
    public void OnCallbackReceived(String ActionID) {
        handler.invoke();
    }
}

File: Sagittarius-Modules/JavaScript/motdmodule.js
var motd;

Sagittarius.prototype.GetMOTD = function () {
	return motd;
};

Sagittarius.prototype.QueryMOTD = function (callback) {
	var ga = this.CreateAction('get'), pass = this.pass;
	this.SubmitAction(
		ga.AddFilter(ga.DBTYPE, 'motd').AddFilter(ga.DBNAME, 'motd').AddProjection('message', true).Unique(),
		function (data) {
			motd = Encryption.Decrypt(data.dbobjects[0].message, pass);
			callback();
		}
	);
};